styx-scheduler-service/src/main/java/com/spotify/styx/docker/LabelValue.java

```diff
@@ -0,0 +1,91 @@
+/*-
+ * -\-\-
+ * Spotify Styx Scheduler Service
+ * --
+ * Copyright (C) 2016 - 2020 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.styx.docker;
+
+import com.google.common.io.BaseEncoding;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.List;
+import java.util.Locale;
+import java.util.regex.Pattern;
+
+final class LabelValue {
+
+  private static final int KUBERNETES_LABEL_MAX_LENGTH = 63;
+  private static final int DIGEST_SUFFIX_LENGTH = 7;
+  private static final int PREFIX_MAX_LENGTH = KUBERNETES_LABEL_MAX_LENGTH - DIGEST_SUFFIX_LENGTH;
+
+  private static final Pattern VALID =
+      Pattern.compile(String.format("^(?:[a-z0-9]|[a-z0-9][a-z0-9_-]{0,%s}[a-z0-9])$",
+          KUBERNETES_LABEL_MAX_LENGTH - 2));
+  private static final Pattern INVALID = Pattern.compile("[^a-z0-9_-]");
+  private static final List<Character> NON_ALPHANUMERICS = List.of('_', '-');
```

Comment:
As we only care about `contains` a `Set` is more appropriate
```suggestion
  private static final Set<Character> NON_ALPHANUMERICS = Set.of('_', '-');
```